dashboard/db/migrate/20240321204728_add_devise_lockable_to_users.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+# Adds all fields required for enabling Devise's Lockable functionality to the
+# User model, in preparation for enabling the feature.
+#
+# See https://github.com/heartcombo/devise/wiki/How-To:-Add-:lockable-to-Users
+class AddDeviseLockableToUsers < ActiveRecord::Migration[6.1]
+  def self.up
+    # Note that we do not add `failed_attempts` or `locked_at` here.
+    # Because neither field is used for querying, neither needs to be an actual
+    # column; we therefore implement both in the properties blob.
+    add_column :users, :unlock_token, :string
+
+    # On MySQL 8.0, adding an index does not lock the table but it does still
+    # take quite a while to apply successfully; see
+    # https://dev.mysql.com/doc/refman/8.0/en/innodb-online-ddl-operations.html#online-ddl-column-syntax-notes
+    #
+    # When tested against a clone of our production database, this operation
+    # took approximately 35 minutes, much longer than the 30 seconds migrations
+    # will wait before erroring out. To avoid this migration failing in
+    # production, we plan to execute the index creation query manually after
+    # the rest of this migration has completed.
```

Comment:
My more well written comment got deleted. I was going to suggest hoisting this bit of information up to the top of the comment. That way our future selves get to read the critical 'what' before digging into the 'why'.